### PR TITLE
API: deprecate Index.sym_diff in favor of symmetric_difference

### LINKED ISSUE #12591: API: rename Index.sym_diff to Index.symmetric_difference
Following builtin `set`, I would expect the method name to be `symmetric_difference`.

How about deprecating `sym_diff` in favour of this (see also #6016, #8226)?

### PR DESCRIPTION
**Pull request by xflr6**
closes #12591

### CODE REVIEW

File: pandas/tests/indexes/common.py [hunk 1]
```diff
@@ -404,7 +404,7 @@ def test_difference_base(self):
                 with tm.assertRaisesRegexp(TypeError, msg):
                     result = first.difference([1, 2, 3])
 
-    def test_symmetric_diff(self):
+    def test_symmetric_difference(self):
```
Comment (jreback) [inline pandas/tests/indexes/common.py]: IIRC there is a test for these types of deprecations, pls add one for ``symmetric_difference``

**Comment by jreback** [comment]
pls add a note in 0.18.1

**Comment by xflr6** [comment]
okay, like this? squash?

**Comment by xflr6** [comment]
is there still something to do here?

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**